ambry-store/src/test/java/com.github.ambry.store/OffsetTest.java

```diff
@@ -0,0 +1,139 @@
+    Offset logOffset = new Offset(name, offset);
+    byte[] serialized = logOffset.toBytes();
+    Offset deserializedOffset =
+        Offset.fromBytes(new DataInputStream(new ByteBufferInputStream(ByteBuffer.wrap(serialized))));
+    assertEquals("Original offset 'name' does not match with the deserialized offset", name,
+        deserializedOffset.getName());
+    assertEquals("Original offset 'offset' does not match with the deserialized offset", offset,
+        deserializedOffset.getOffset());
+    // equals test
+    assertEquals("Original offset does not match with the deserialized offset", logOffset, deserializedOffset);
+    // hashcode test
+    assertEquals("Hashcode doesn't match", logOffset.hashCode(), deserializedOffset.hashCode());
+  }
+
+  /**
+   * Tests the constructor and {@link Offset#fromBytes(DataInputStream)} function with bad input.
+   * @throws IOException
+   */
+  @Test
+  public void offsetBadInputTest()
+      throws IOException {
+    doBadOffsetInputTest(null, 10);
+    doBadOffsetInputTest("", 10);
+    doBadOffsetInputTest("1_11_log", -1);
+
+    Offset offset = new Offset("1_11_log", 10);
+    byte[] serialized = offset.toBytes();
+    // mess with a version byte
+    serialized[0] = serialized[0] == (byte) 1 ? (byte) 2 : (byte) 1;
+    try {
+      Offset.fromBytes(new DataInputStream(new ByteBufferInputStream(ByteBuffer.wrap(serialized))));
+      fail("Version check should have failed");
+    } catch (IllegalArgumentException e) {
+      // expected.
+    }
+  }
+
+  /**
+   * Tests {@link Offset#compareTo(Offset)}.
+   */
+  @Test
+  public void compareToTest() {
+    // TODO (Log Segmentation): Improve and add more cases once compareTo also uses the name.
+    Offset lower = new Offset("lower", 1);
+    Offset match = new Offset("match", 1);
+    Offset higher = new Offset("higher", 2);
+    assertEquals("CompareTo result is inconsistent", -1, lower.compareTo(higher));
+    assertEquals("CompareTo result is inconsistent", 0, lower.compareTo(match));
+    assertEquals("CompareTo result is inconsistent", 1, higher.compareTo(lower));
+  }
+
+  /**
+   * Tests for {@link Offset#equals(Object)} and {@link Offset#hashCode()}.
+   */
+  @Test
+  public void equalsAndHashCodeTest() {
+    Offset o1 = new Offset("test", 2);
+    Offset o2 = new Offset("test", 2);
+    Offset o3 = new Offset("test", 3);
+    Offset o4 = new Offset("test_more", 2);
+
+    assertTrue("Offset should be equal to itself", o1.equals(o1));
+    assertFalse("Offset should not be equal to null", o1.equals(null));
+    assertFalse("Offset should not be equal to random Object", o1.equals(new Object()));
+
+    assertTrue("Offsets should be declared equal", o1.equals(o2));
+    assertEquals("Hashcode mismatch", o1.hashCode(), o2.hashCode());
+
+    assertFalse("Offsets should be declared unequal", o1.equals(o3));
+    assertTrue("Hashcode should be declared unequal", o1.hashCode() != o3.hashCode());
```

Comment:
as discussed in the other PR, let us remove this check - simply because if this fails it does not imply that there is a correctness issue.